src/redux/note-content/reducers.ts

```diff
@@ -0,0 +1,127 @@
+/*
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import { DateTime } from 'luxon'
+import { Reducer } from 'redux'
+import { Note } from '../../api/notes'
+import { YAMLMetaData } from '../../components/editor/yaml-metadata/yaml-metadata'
+import {
+  NoteContent,
+  NoteContentAction,
+  NoteContentActionType,
+  SetCheckboxInMarkdownContentAction,
+  SetNoteContentAction,
+  SetNoteDataFromServerAction,
+  SetNoteMetaDataFromRenderingAction,
+  UpdateNoteTitleByFirstHeadingAction
+} from './types'
+
+export const initialState: NoteContent = {
+  markdownContent: '',
+  id: '',
+  createTime: DateTime.fromSeconds(0),
+  lastChange: {
+    timestamp: DateTime.fromSeconds(0),
+    userId: ''
+  },
+  alias: '',
+  preVersionTwoNote: false,
+  viewCount: 0,
+  authorship: [],
+  noteTitle: '',
+  firstHeading: '',
+  metadata: {
```

Comment:
Maybe call it `yamlMetadata` to make clear that this is the frontmatter defined data?

Reply:
Then it would be "frontmatterMetadata" or just "frontmatter", because I don't know if we want the purpose to the original format.

Reply:
I'm fine with `frontmatter`.